logstash-rs/src/error.rs

```diff
@@ -0,0 +1,19 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
```

Comment:
I would think about some other errors. Because as user of the library you don't receive any valuable information now. What would you do if you get `LockStreamMutex` error?